LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -678,9 +678,31 @@ EtwMonitor::OnRecordEvent(
 
         if (ERROR_SUCCESS != status)
         {
+            // Handle 1168 fails in this block
+            if (ERROR_NOT_FOUND == status)
+            {
+                // Access the ProviderId from the EVENT_HEADER field
+                GUID providerId = EventRecord->EventHeader.ProviderId;
+
+                LPOLESTR clsidString;
+                if (StringFromCLSID(providerId, &clsidString) != S_OK)
+                {
+                    logWriter.TraceError(
+                        Utility::FormatString(L"Failed to allocate necessary memory Error: %lu",
+                        E_OUTOFMEMORY).c_str());
+                }
+
+                std::wstring guidString(clsidString);
+                CoTaskMemFree(clsidString);
+
+                logWriter.TraceError(
+                    Utility::FormatString(L"Failed to query ETW event information for ProviderGUID: %s Error: %lu",
+                    guidString, status).c_str());
+            }
+
+            // Handle all non 1168 fails here
             logWriter.TraceError(
-                Utility::FormatString(L"Failed to query ETW event information. Error: %lu", status).c_str()
-            );
+                Utility::FormatString(L"Failed to query ETW event information. Error: %lu", status).c_str());

```

Comment:
All failures should report the failing provider id

Reply:
We can get rid of the if (ERROR_NOT_FOUND == status) check and have this logic for all the failing providers.